### PR TITLE
Pageserver read trace utils

### PR DESCRIPTION
**Pull request by bojanserafimov**
List, dump, and analyze read traces. Replay not implemented yet, I'm saving it for a separate PR. It will be complicated, since it might involve simultaneous wal replay.

The analysis is very primitive but it can be used to detect issues like https://github.com/neondatabase/neon/issues/2585

### CODE REVIEW

**Comment by bojanserafimov** [comment]
Not sure why tests are failing `fatal: detected dubious ownership in repository at` ...

File: trace/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    // Compute stats
+    while let Ok(msg) = PagestreamFeMessage::parse(&mut reader) {
+        match msg {
+            PagestreamFeMessage::Exists(_) => {}
+            PagestreamFeMessage::Nblocks(_) => {}
+            PagestreamFeMessage::GetPage(req) => {
+                total += 1;
+
+                let delta = (req.blkno as i32) - (prev as i32);
```
[the remark this answers is not in the document]
Comment (bojanserafimov) [reply trace/src/main.rs:65]: They're rare so I didn't bother. I'll address or leave a `HACK` comment. Thanks for pointing out.

**Comment by bojanserafimov** [comment]
Test failure is not transient:
```
INFO[2164] Running: [/bin/sh -c cd compute_tools && cargo build --locked --profile release-line-debug-size-lto] 
error: failed to load manifest for workspace member `/home/nonroot/trace`

Caused by:
  failed to read `/home/nonroot/trace/Cargo.toml`

Caused by:
  No such file or directory (os error 2)
error building image: error building stage: failed to execute command: waiting for process to exit: exit status 101
Error: Process completed with exit code 101.
```

Seems like a dockerfile problem. @SomeoneToIgnore does the problem seem obvious? (I found you by git blame)

File: trace/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                prev = req.blkno;
+
+                match deltas.get_mut(&delta) {
```
[the remark this answers is not in the document]
Comment (koivunej) [reply trace/src/main.rs]: Late to this party but:

```rust
*deltas.entry(&delta).or_default() += 1;
```

Is smaller :)

**Pull request merged**